core/src/main/java/io/questdb/griffin/engine/functions/math/LeastNumericFunctionFactory.java

```diff
@@ -181,14 +191,17 @@ public ObjList<Function> getArgs() {
         @Override
         public long getLong(Record rec) {
             long value = Long.MAX_VALUE;
+            boolean allAreNull = true;
```

Comment:
this is naive logic, please refer to questdb.com front page slogan :)

faster and simpler logic would be:

```java
            long value = args.getQuick(0).getLong(rec);
            for (int i = 1; i < n; i++) {
                value = Math.max(value, args.getQuick(i).getLong(rec));
            }
            return value;
```

the `n` should be cached to allow JVM unroll small loops

Reply:
This loop won't work for the least function. The difference between this loop and what's in greatest()'s long override is minus one branch.

Reply:
yeah, that's what i said just above :)

Reply:
yes, in practice we are IO bound, it but doesn't mean we should be writing CPU-inefficient code and giving up on CPU optimisation.